### DISCUSSION TITLE
highlight all matching texts when searching

### DISCUSSION THREAD

**Discussion by tksoh** [eteran/nedit-ng#213] [Ideas]
Many modern editors, including Qt Creator, would highlight all the texts in the document that are matching the search string. In fact, Notepad++ takes it one step further by highlighting all matching text when we select any word in the editing window. Also, Qt Creator, like google Chrome, would add indicators on the scroll bar to mark the location of matching texts, which also provide the visual que on how many matches there are. Whichever way, this feature greatly assist the searching of text on large text files.

I tried to implement in nedit 5.x years ago, but it was a little too resource intensive to be practical for large files. Perhaps Qt will make things easier.

**Comment by eteran** [comment]
Yup, this has been on my radar for a bit too. I considered implementing it in terms of rangesets, but for performance reasons, we may need something more hand tailored.
